tests/integration/targets/azure_rm_sqlserver/tasks/main.yml

```
@@ -67,6 +166,58 @@
       - output.servers.sqlsrv{{ random_postfix }}.state != None
       - output.servers.sqlsrv{{ random_postfix }}.fully_qualified_domain_name != None
       - output.servers.sqlsrv{{ random_postfix }}.tags.aaa == 'bbb'
+      - output.servers.sqlsrv{{ random_postfix }}.admin_username == 'mylogin'
+      - output.servers.sqlsrv{{ random_postfix }}.administrators == None
+
+- name: Gather facts for extended SQL Server
+  azure_rm_sqlserver_info:
+    resource_group: "{{ resource_group }}"
+    server_name: "sqlsrv-extended-{{ random_postfix }}"
+  register: output
+- name: Assert that facts are returned
+  assert:
+    that:
+      - output.changed == False
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].id != None
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].name == 'sqlsrv-extended-{{ random_postfix }}'
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].type != None
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].location != None
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].kind != None
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].version != None
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].state != None
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].fully_qualified_domain_name != None
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].tags.aaa == 'bbb'
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].minimal_tls_version == '1.2'
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].public_network_access == 'Disabled'
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].restrict_outbound_network_access == 'Enabled'
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].admin_username == 'mylogin2'
+      - output.servers['sqlsrv-extended-{{ random_postfix }}'].administrators == None
+
+- name: Gather facts for SQL Server with Azure AD admin
+  azure_rm_sqlserver_info:
+    resource_group: "{{ resource_group }}"
+    server_name: "sqlsrv-azuread-{{ random_postfix }}"
+  register: output
+- name: Assert that facts are returned
+  assert:
+    that:
+      - output.changed == False
+      - output.servers['sqlsrv-azuread-{{ random_postfix }}'].id != None
```

In YAML, variable values should be referenced with ".".  Change everything like this. Thank you very much!
```suggestion
      - output.servers.sqlsrv-azuread-{{ random_postfix }}.id != None
```

---

The output format of having `servers` be a list of dictionaries is existing functionality (see same in [master branch](https://github.com/l3ender/azure/blob/094fff8449a1e39deecc9945827f90c42cf7a8d7/plugins/modules/azure_rm_sqlserver_info.py#L51)).